src/tools/queryDatasource/validators/validateFilterValues.ts

```diff
@@ -41,15 +43,15 @@ export async function validateFilterValues(
       (filter.filterType === 'SET' || filter.filterType === 'MATCH') &&
       'fieldCaption' in filter.field &&
       filter.field.fieldCaption,
-  );
+  ) as Array<z.infer<typeof MatchFilter> | z.infer<typeof SetFilter>>;
```

Comment:
Feel free to add types for both of these so we don't have to use `z.infer` all the time.

`export type SetFilter = z.infer<typeof SetFilter>;`